app/models/user.rb

```diff
@@ -0,0 +1,42 @@
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  has n, :peeps
+
+  property :id, Serial
+  property :email, String, unique: true
+  property :password_digest, Text
+  property :password_token, Text
+  property :password_token_timestamp, Time
+  property :name, String
+  property :username, String, unique: true
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_uniqueness_of :email
+  validates_uniqueness_of :username
+  validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.